### DISCUSSION TITLE
Examples of Action policy usage for learning purposes

### DISCUSSION THREAD

**Discussion by Shamaoke** [palkan/action_policy#275] [Q&A]
Hi.

I need to implement user authentication and authorization in a Rails application I'm developing. For that purpose I decided to use the Action policy gem. I'm completely new with this tool and I'd like to get familiar with its basics.  Since, I believe, learning is best done by examples, I'm looking for code of a demo Rails application to understand what things are needed, where they should be located and how they interact with each other in making authentication and authorization possible.

The application I'm looking for should be a standard Rails application that includes Action policy as a component. There shouldn't be any components that work together with Action policy to achieve the task or any other components that might complicate the matter of learning.

Any documentation or articles that consistently describe the implementation of such an application would be a big plus.

Do you know any complete examples of Rails applications intended or could be intended to familiarize you with Action policy usage?

Thanks.

**Comment by ezekg** [comment]
My company uses it extensively, and we're fair source on GitHub: https://github.com/keygen-sh/keygen-api/tree/master/app/policies

**Comment by palkan** [comment]
I don't have any of my work with Action Policy open-sourced (yet), but you can check the dependency graph [here](https://github.com/palkan/action_policy/network/dependents) and see the projects using Action Policy. There are some notable examples, such as:
- (already mentioned) https://github.com/keygen-sh/keygen-api
- https://github.com/rubyforgood/pet-rescue
- https://github.com/kortirso/pullmetry
- https://github.com/TelosLabs/rails-world
- https://github.com/MarkUsProject/Markus
- https://github.com/changepack/changepack
